src/solvers/mp/rcspp/RCGraph.cpp

```diff
@@ -0,0 +1,1095 @@
+            l_alloc.allocate( 1 );
+        l_alloc.construct( new_label,
+                           boost::r_c_shortest_paths_label
+                               <Graph, Resource_Container>
+                               ( i_label_num++,
+                                 cur_label->cumulated_resource_consumption,
+                                 cur_label.get(),
+                                 *oei,
+                                 target( *oei, g ) ) );
+        nbCreatedLabels++;
+
+        b_feasible =
+            ref( g,
+                 new_label->cumulated_resource_consumption,
+                 new_label->p_pred_label->cumulated_resource_consumption,
+                 new_label->pred_edge );
+
+        if( !b_feasible )
+        {
+          vis.on_label_not_feasible( *new_label, g );
+          new_label->b_is_valid = false;
+          l_alloc.destroy( new_label );
+          l_alloc.deallocate( new_label, 1 );
+          nbDeletedLabels++;
+        }
+        else
+        {
+          const boost::r_c_shortest_paths_label<Graph, Resource_Container>&
+              ref_new_label = *new_label;
+          vis.on_label_feasible( ref_new_label, g );
+          Splabel new_sp_label( new_label );
+          vec_vertex_labels[new_sp_label->resident_vertex].
+              push_back( new_sp_label );
+          unprocessed_labels.push( new_sp_label );
+        }
+      }
+    }
+    else
+    {
+      assert (cur_label->b_is_valid);
+      vis.on_label_dominated( *cur_label, g );
+      cur_label->b_is_valid = false;
+      l_alloc.destroy( cur_label.get() );
+      l_alloc.deallocate( cur_label.get(), 1 );
+      nbDeletedLabels++;
+    }
+  }
+
+  // ------------------------------------------------------------------------- START SAMUEL
+  typename std::list<Splabel>::const_iterator csi;
+  typename std::list<Splabel>::const_iterator csi_end;
+  for(int sink=0; sink<t.size(); sink++){
+    std::list<Splabel> dsplabels = get(vec_vertex_labels, t[sink]);
+    csi = dsplabels.begin();
+    csi_end = dsplabels.end();
+    // if d could be reached from o
+    if( !dsplabels.empty() )
+    {
+      for( ; csi != csi_end; ++csi )
+      {
+        std::vector<typename boost::graph_traits<Graph>::edge_descriptor>
+            cur_pareto_optimal_path;
+        const boost::r_c_shortest_paths_label<Graph, Resource_Container>* p_cur_label =
+            (*csi).get();
+        assert (p_cur_label->b_is_valid);
+        pareto_optimal_resource_containers.
+            push_back( p_cur_label->cumulated_resource_consumption );
+        while( p_cur_label->num != 0 )
+        {
+          cur_pareto_optimal_path.push_back( p_cur_label->pred_edge );
+          p_cur_label = p_cur_label->p_pred_label;
+          assert (p_cur_label->b_is_valid);
+        }
+        pareto_optimal_solutions.push_back( cur_pareto_optimal_path );
+        if( !b_all_pareto_optimal_solutions )
+          break;
+      }
+    }
+  }
+  /*
```

Comment:
delete comment below?

Reply:
It's boost original code. I would keep it.